corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -600,13 +600,23 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
             var fetchingPrompts = FormplayerFrontend.getChannel().request("app:select:menus", urlObject);
             $.when(fetchingPrompts).done(function (response) {
                 // Update models based on response
-                _.each(response.models, function (responseModel, i) {
-                    self.collection.models[i].set({
-                        error: responseModel.get('error'),
-                        required: responseModel.get('required'),
-                        required_msg: responseModel.get('required_msg'),
-                    });
+                if (response.queryResponse != null) {
```

Comment:
yep correct only responses of type entities have the queryResponse and that only occurs on the Formplayer side with the FF enabled

Reply:
*Should* these displays be backbone models instead of plain js objects? I wonder if the right way to fix this is in initialization code, and if there could be other side effects from the difference in data type (maybe not bugs, since you're accessing the displays differently in the two cases, but for readability & comprehensibility it would be nice to have consistent code).

That said, it's fine for understanding that question to be followup work.